### DISCUSSION TITLE
Request: Official support for: Setup Xcode version in GitHub Action

### DISCUSSION THREAD

**Discussion by salvatoreboemia** [community/community#160490] [Actions]
Hi, browsing the marketplace I found this extension

Is it possible to integrate it directly into github without having to use a third-party extension?

In addition to this for example

```
     with:
       macos-version: ${{ vars.MACOSVERSION}}
```

Also specify an 

```
xcode-version: ${{ vars.XCODEVERSION}}
```

That would be super helpful

Or at least have the code or a guide on how to specify the Xcode version easily in a GitHub Reusable Action so we can use it in our workflow

Probably the right code is this one, but i'm not sure:
```
- name: Select Xcode Version
   run: sudo xcode-select -switch /Applications/Xcode_${XCODEVERSION}.app
```

Thanks for help

**Comment by vpolikarpov-akvelon** [comment]
Hey @salvatoreboemia. I found this actions: https://github.com/maxim-lobanov/setup-xcode. Is it what you are looking for?

**Comment by salvatoreboemia** [comment]
> Hey @salvatoreboemia. I found this actions: https://github.com/maxim-lobanov/setup-xcode. Is it what you are looking for?

Yes exactly. But I wouldn't want to use a third-party action.

Would it be possible to integrate this directly into GitHub as a feature?

Or know the exact code to do it manually and without third-party actions?

Thanks

**Reply by vpolikarpov-akvelon** [reply]
Well, in fact no actions are integrated in GitHub as a feature. Even a fundamental ones like [checkout](https://github.com/actions/checkout) are provided as a repo. The only difference is an organization that owns repository. As of [setup-xcode](https://github.com/maxim-lobanov/setup-xcode), it's unlikely that it will be moved anywhere. But I also want to mention that it's author is a member of @github org so it's not so 3rd-party as you may think. Anyway you can fork repository for personal use if you want.

**Reply by vpolikarpov-akvelon** [reply]
If you want to know how aforementioned action works you may examine its [source code](https://github.com/maxim-lobanov/setup-xcode/tree/master/src).

**Comment by FlorinDMass** [comment]
You can use : 
`- name: Set Xcode version
      run: sudo xcode-select -s /Applications/Xcode_16.2.app`
